migrate_nidirect_node/migrate_nidirect_node_nidirect_contact/src/Plugin/migrate/source/NIDirectContactNodeSource.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Drupal\migrate_nidirect_node_nidirect_contact\Plugin\migrate\source;
+
+use Drupal\Core\Entity\EntityManagerInterface;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\State\StateInterface;
+use Drupal\migrate\Plugin\MigrationInterface;
+use Drupal\node\Plugin\migrate\source\d7\Node;
+use Drupal\migrate\Row;
+use Drupal\migrate_nidirect_utils\TelephonePlusUtils;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Logger\LoggerChannelFactory;
+
+/**
+ * Prepares NIDirect Contact nodes for telephone plus field.
+ *
+ * @MigrateSource(
+ *   id = "nidirect_contact_node_source",
+ * )
+ */
+class NIDirectContactNodeSource extends Node implements ContainerFactoryPluginInterface {
+
+  /**
+   * Logger.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelFactory
+   */
+  protected $logger;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, MigrationInterface $migration, StateInterface $state, EntityManagerInterface $entity_manager, ModuleHandlerInterface $module_handler, LoggerChannelFactory $logger) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $migration, $state, $entity_manager, $module_handler);
+    $this->logger = $logger->get('NIDirectContactNodeSource');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition, MigrationInterface $migration = NULL) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $migration,
+      $container->get('state'),
+      $container->get('entity.manager'),
+      $container->get('module_handler'),
+      $container->get('logger.factory')
```

Comment:
Seems to me that you're only using the logger here, do we need the entity manager and module handler ?

Reply:
Ah, Good catch Martin. I originally did everything in this class which required those but then spun it out to the Utils class.

Reply:
Although, the parent class calls these, so I'm gonna leave them in.